### PR TITLE
Speed up registry doctest

### PR DESCRIPTION
**Pull request by msdemlei**
This is using other constraints and services in order to transfer less data and perhaps have faster queries.

### CODE REVIEW

File: docs/registry/index.rst [hunk 1]
```diff
@@ -34,7 +34,9 @@ Basic interface
 
 
 The main interface for the module is :py:meth:`pyvo.registry.search`;
-the examples below assume::
+the examples below assume:
+
+.. doctest::
```
Comment (bsipocz) [inline docs/registry/index.rst:39]: no need for this directive, code with `>>>` are picked up for doctesting
```suggestion
```

**Review by bsipocz — APPROVED** [review]
Minor comment, otherwise looks good.

Thanks!

**Comment by bsipocz** [comment]
while checking this locally, I discovered some weird behaviour/bug of the tests, I'll open a separate issue for that as the behaviour is present on `main`, too.

**Comment by msdemlei** [comment]
On Fri, Jun 28, 2024 at 09:58:34AM -0700, Brigitta Sipőcz wrote:
> @bsipocz approved this pull request.
>
> Minor comment, otherwise looks good.
>
> Thanks!
>
> > +
> +.. doctest::
>
> no need for this directive, code with `>>>` are picked up for doctesting
> ```suggestion
> ```

How bad is the spurious doctest::?  I'm asking because to simplify
the logic of my doctest profiler,
https://blog.g-vo.org/watch-sphinx-doctests.html, I'm only picking up
python lines from RST blocks with doctest directives.

I *can* loosen this up, but if the occasional explicit but
unnecessary directive doesn't bother anyone badly, I'd like it better
if were were explicit.

[And yes, it would be preferable if I used whatever logic sphinx uses
to pick up the tests for the profiler, but I figured I'd spend a lot
more time on locating that than I wanted to spend on this
tooling-for-tooling thing].

**Comment by bsipocz** [comment]
We don't use sphinx for picking up doctests and its plumbing is somewhat different than the pytest ecosystem and plugins we use (e.g. some sphinx doctest directives are known to not work with pytest/pytest-doctestplus and similarly, some doctestplus things that we rely on are known not working with sphinx.). So, while most things  may work without issues, I wouldn't be surprised if you see doctest failures in CI even if they pass with the sphinx system locally.

So, having this one `.. doctest::` is simply superfluous but doesn't break anything, so can stay, but you will need to keep using `>>>` for the code snippets otherwise they won't be picked up by the system we use, even though sphinx doctest may be happy with just using the doctest directive.

**Review by andamian — APPROVED** [review]

**Pull request merged**